core/trino-main/src/main/java/io/trino/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -349,87 +357,37 @@ public Context(
             this.timeoutInMilliseconds = timeoutInMilliseconds;
             this.session = session;
             this.warningCollector = warningCollector;
+            this.planOptimizersStatsCollector = planOptimizersStatsCollector;
+            this.iterativeOptimizerStatsCollector = new PlanOptimizersStatsCollector(5);
```

Comment:
Having two collectors and recording stats twice is not great. We should figure out how to consolidate (by having the error reporting below use the provided collector instead, by moving the error reporting elsewhere, or by removing it altogether)

Reply:
Removed duplicate recording of statistics.

Reply:
@martint could you PTAL and drop the request changes if it's okay now ?